### ISSUE TITLE
dafny format on comments

### ISSUE THREAD

**Issue by davidcok** [dafny-lang/dafny#3524]
### Summary

It appears dafny format does nothing with comments except to align the opening '*'. This case however merits attention:

```
  /* The quick brown fox
jumped over the lazy dog */
```

### Background and Motivation

To make the best formatter ever.

### Proposed Feature

I suggest that in the case of a block comment in which 
a) the subsequent lines do not start with a *
b) and extend further to the left than the first line
that the subsequent lines be moved right to line up the first non-space characters (but not ever moved left)

So
```
   /* The quick brown fox
  jumped over 
the lazy dog. */
```

becomes
```
   /* The quick brown fox
      jumped over 
      the lazy dog. */
```

But

```
   /*  The quick brown fox
  jumped over 
 *    the lazy dog. */
```

becomes

```
   /*  The quick brown fox
       jumped over 
    *    the lazy dog. */
```

**Comment by MikaelMayer** [comment]
I considered doing that, and I decided we should not (I'll explain with a few examples)
I decided the invariant to be that whatever is between /* and */, if the first /* is not followed by a line starting with a "*", the entire comment should be kept as a block. Indentation is changed only if `/*` is on its own line, and even then, the relative columns of everything between the `/*` ... `*/`is unchanged.
That gives at least one clear way to temporarily comment code and not have everything be messed up.

For example, what about:

```dafny
         case C =>
           D()  /*
         case A =>
           B()
       }
       print "Hello"
  */
       }
```

With your style, it would completely mess up with the indentation of the temporarily commented out code.
Multiline comments are used for too many things like that to even try messing up with the internal indentation or alignment.
The only exception I tolerated is for all the consecutive lines that start with a "*"
So

```
  const a := 1;
  /* a is a constant
* that should not be a prime number */
```
is realigned into
```
  const a := 1;
  /* a is a constant
   * that should not be a prime number */
```
